logstash-core/lib/logstash/api/commands/default_metadata.rb

```diff
@@ -20,7 +20,9 @@ def version
         end
 
         def http_address
-          service.agent.webserver.address
+          @http_address ||= service.get_shallow(:http_address).value
+        rescue ::LogStash::Instrument::MetricStore::MetricNotFound, NoMethodError => e
+          nil
```

Comment:
@ph in testing, unfortunately, the metric subsystem isn't initialized, so service is nil